api/nakadi-event-bus-api.yaml

```diff
@@ -623,6 +539,358 @@ paths:
+          description: |
+            Id of stream which client uses to read events. It is not possible to make a commit for a terminated or
+            none-existing stream. Also the client can't commit something which was not sent to his stream.
+        - name: cursors
+          in: body
+          schema:
+            type: object
+            properties:
+              cursors:
+                description: |
+                  List of cursors that the consumer acknowledges to have successfully processed.
+                type: array
+                items:
+                  $ref: '#/definitions/SubscriptionCursor'
+            required:
+              - stream_id
+              - items
+      responses:
+        '204':
+          description: Offsets were committed
+        '200':
+          description: |
+            At least one cursor which was tried to be committed is older or equal to already committed one. Array
+            of commit results is returned for this status code.
+          schema:
+            type: object
+            properties:
+              items:
+                description: list of items which describe commit result for each cursor
+                type: array
+                items:
+                  $ref: '#/definitions/CursorCommitResult'
+            required:
+              - items
+        '404':
+          description: Subscription not found
+          schema:
+            $ref: '#/definitions/Problem'
+        '422':
+          description: Unprocessable Entity
+          schema:
+            $ref: '#/definitions/Problem'
+        '429':
+          description: Too many requests
+          schema:
+            $ref: '#/definitions/TooManyRequests'
+
+  /subscriptions/{subscription_id}/events:
+    get:
+      tags:
+        - subscription-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: |
+        Starts a new stream for reading events from this subscription. The data will be automatically rebalanced
+        between streams of one subscription. The minimal consumption unit is a partition, so it is possible to start as
+        many streams as the total number of partitions in event-types of this subscription. The rebalance currently
+        only operates with the number of partitions so the amount of data in event-types/partitions is not considered
+        during autorebalance.
+        The position of the consumption is managed by Nakadi. The client is required to commit the cursors he gets in
+        a stream.
+      parameters:
+        - $ref: '#/parameters/SubscriptionId'
+        - name: max_uncommitted_events
+          in: query
+          description: |
+            The amount of uncommitted events Nakadi will stream before pausing the stream. When in paused
+            state and commit comes - the stream will resume. Minimal value is 1.
+          type: integer
+          format: int32
+          required: false
+          default: 10
```

Comment:
Imagine a client is able to process 10k events per second. If the client doesn't change this default value, it really doesn't matter if it's going to be 10 or 100. It will be at least 3 orders of magnitude under it's maximum throughput capability. So the default value will be bad anyway.

In my opinion, this value should be dynamic. But we can add this in the future.